lighthouse-core/audits/dobetterweb/no-datenow.js

```diff
@@ -61,17 +61,26 @@ class NoDateNowAudit extends Audit {
       });
     }
 
+    let debugString;
+
     const pageHost = new URL(artifacts.URL.finalUrl).host;
     // Filter usage from other hosts and keep eval'd code.
-    // If there is no .url in the violation, include it in the results because
-    // we cannot determine if it was from the user's page or a third party.
-    // TODO: better extendedInfo for violations with no URL.
-    // https://github.com/GoogleChrome/lighthouse/issues/1263
     const results = artifacts.DateNowUse.usage.filter(err => {
       if (err.isEval) {
         return !!err.url;
       }
-      return err.url ? new URL(err.url).host === pageHost : true;
+
+      // If the violation doesn't have a valid url, don't filter it out, but
+      // warn the user that we don't know what the callsite is.
```

Comment:
one could argue this is ripe for a helper refactor, but we only have 2 atm. Don't think that warrants it just yet.